src/workerd/api/actor-state.h

```diff
@@ -342,6 +419,11 @@ class DurableObjectTransaction final: public jsg::Object, public DurableObjectSt
 
   bool rolledBack = false;
 
+  // Called from C++ after the transaction callback has completed (successfully or not). These
+  // methods do nothing if the transaction is already committed / rolled back.
+  kj::Promise<void> maybeCommit();
+  void maybeRollback();
```

Comment:
I'm curious why we had to make these methods private?  Doing so requires us to use a friend class internally?

Reply:
It's mostly about maintainability for me. Since these are private, we can be certain that we control each spot in which they are called. That means that I don't have to worry about them leaking exceptions across the api boundary since they are only called from functions that also should enforce the api boundary.